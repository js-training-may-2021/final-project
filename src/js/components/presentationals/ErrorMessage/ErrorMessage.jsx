import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({message}) => {
	return (
		<p className="page__text page__text--message">{message}</p>
	);
};

ErrorMessage.propTypes = {
	message: PropTypes.string.isRequired,
};

export default React.memo(ErrorMessage);
