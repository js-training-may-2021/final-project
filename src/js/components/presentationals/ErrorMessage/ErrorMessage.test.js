import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './ErrorMessage.jsx';

it(`Render ErrorMessage`, () => {
  const tree = renderer
    .create(
      <ErrorMessage
        message={'Failed to load caught pokemons. Please try again later.'}
      />)
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children[0]).toBe('Failed to load caught pokemons. Please try again later.');
});
