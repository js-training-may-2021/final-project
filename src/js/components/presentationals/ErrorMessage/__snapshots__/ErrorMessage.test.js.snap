// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ErrorMessage 1`] = `
<p
  className="page__text page__text--message"
>
  Failed to load caught pokemons. Please try again later.
</p>
`;
